py_hamt/sharded_zarr_store.py

```
@@ -0,0 +1,870 @@
+
+    async def _load_root_from_cid(self):
+        root_bytes = await self.cas.load(self._root_cid)
+        try:
+            self._root_obj = dag_cbor.decode(root_bytes)
+            if not isinstance(self._root_obj, dict) or "chunks" not in self._root_obj:
+                raise ValueError(
+                    "Root object is not a valid dictionary with 'chunks' key."
+                )
+            if not isinstance(self._root_obj["chunks"]["shard_cids"], list):
+                raise ValueError("shard_cids is not a list.")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)